### ISSUE TITLE
Automatic clan castle troop calculation not flexible enough

### ISSUE THREAD

**Issue by raysun** [raysun/Clash-of-Clans-Notifier#6]
Worked for my clan, but some clans mix up the cc troops & have multiple donators. 

other features:
* assign who should donate to the highest level people. but does that really work - what if those players are not available or they are slow? and what if you don't need the absolute highest level troops but would rather have them quickly?

**Comment by xDp64xG** [comment]
So @ the highest person with the troops for the war cc troops, to improve the waiting time? Bot works by figuring out which troops they want to donate, and search for the highest troop lvls, and alerts them via discord

**Comment by raysun** [comment]
yep, @ is possible. the broader question is should i write the UI for customization. THx = what troops. or if this even matters, since some clans mix up the troops intentionally.

**Comment by raysun** [comment]
This feature is completely different now. But there's still a question as to a) why should a strong clan trust our war cc recommendation? and b) What levers do they have to tweak (e.g. for some reason, they don't ever want to use troop X and Y)

**Comment by Peter-AC** [comment]
So, (A), a strong clan was never the target market for this feature. It would take a TON of time to bake down the fuzzy intelligence of Clash experts to reproduce what a strong clan does all the time. I don't think that's a worthwhile goal.

(B) is where it gets more complicated, because it starts to address the middle of the spectrum. Neither terrible nor superstar clans. Honestly, the first step is to weed out lesser combos from cc_troops.json and adjust the probabilities with the help of a Clash expert. That is the easiest change to make. Past that, I would need to know where and how the current algorithm is lacking to determine better handles for controlling or directing the randomness.

Even then, we would need to favor handles for experts to craft or handles for beginners to not get screwed. I suspect that the two conflict with each other.

The original issue, that of setting who donates to whom, is a mess. There are all sorts of donation strategies, and we could spend months getting blindsided by feature requests for more control. Or we could come up with a genius solution that works for darn near everyone easily. Who knows.

**Comment by raysun** [comment]
Right. I'll move this to v2 and we can consider a broader change then.
